node/src/components/consensus/era_supervisor.rs

```diff
@@ -529,12 +529,12 @@ where
             let start_height;
             let era_start_time;
 
-            if era_id.is_genesis() {
+            if let Some(start_time) = genesis_start_time {
```

Comment:
That still needs to check whether `era_id.is_genesis()`, otherwise it uses this start time in _all_ iterations.

Reply:
Ah - good catch!

I guess I would expect a test to catch this case.  I even ran nctl with this code and it got past era 0.

Reply:
Yes, I would have expected that, too: Do we have a test where a node joins in an era other than 1, but without an upgrade?

Reply:
We do but it runs only in nightly build so it would break there.